### PR TITLE
New `utteranc` comments system

### PR DESCRIPTION
**Pull request by TrumanDu**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
- [x] Muse | Mist have been tested.
- [ ] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [x] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A
[Should notice user about the permission of gitment ](https://github.com/theme-next/hexo-theme-next/issues/35)
## What is the new behavior?

add new comments 
- Screenshots with this changes: N/A
QQ截图20190330235029

- Link to demo site with this changes: N/A
[demo](http://trumandu.github.io/about/)
### How to use?
In NexT `_config.yml`:
```yml
# utteranc
# Demo: https://utteranc.es/  http://trumandu.github.io/about/
# theme: github-light,github-dark,github-dark-orange
utteranc:
  enable: true
  repo:  TrumanDu/comments
  pathname:  pathname
  theme: github-light
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

**Comment by LEAFERx** [comment]
WOW! Seems a little better than gitment.

**Comment by TrumanDu** [comment]
ready now?

**Comment by JiangTJ** [comment]
@ivan-nginx @TrumanDu 
There are some controversies about reviews, which can take a long time, so we can validate and merge the PR first.

**Comment by ivan-nginx** [comment]
@JiangTJ after that merge u will reedit your PR? If yes, of course merge it.

**Comment by JiangTJ** [comment]
> @JiangTJ after that merge u will reedit your PR? If yes, of course merge it.

Yesh, I will reedit my PR after this merge.

File: _config.yml [hunk 1]
```diff
@@ -679,6 +679,14 @@ gitalk:
   # Available values: en, es-ES, fr, ru, zh-CN, zh-TW
   language:
 
+# utteranc
+# Demo: https://utteranc.es/  http://trumandu.github.io/about/
+# theme: github-light,github-dark,github-dark-orange
```
Comment (stevenjoezhang) [inline _config.yml]:
```
# Demo: https://utteranc.es/, http://trumandu.github.io/about/
# theme: github-light, github-dark, github-dark-orange
```

**Comment by ivan-nginx** [comment]
@TrumanDu please, add docs to NexT website.

**Comment by ivan-nginx** [comment]
@TrumanDu please, add docs to NexT website.

**Comment by TrumanDu** [comment]
@ivan-nginx [here ](https://github.com/theme-next/theme-next.org/pull/55)

**Comment by 1v9** [comment]
@TrumanDu Take a look at requested change, once done we can merge this PR.

File: layout/_third-party/comments/utteranc.swig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{% set utteranc_js_url = '//utteranc.es/client.js' %}
+
+<script src="https:{{utteranc_js_url}}"
+        repo="{{ theme.utteranc.repo }}"
+        issue-term="{{ theme.utteranc.pathname }}"
+        theme="{{ theme.utteranc.theme }}"
+        crossorigin="anonymous"
+        async>
+</script>
```
Comment (1v9) [inline layout/_third-party/comments/utteranc.swig]:
```
<script src="https:{{utteranc_js_url}}"
  repo="{{ theme.utteranc.repo }}"
  ...
  async>
</script>

File: _config.yml (hunk 1 above)
Comment (1v9) [inline _config.yml]: You missed three themes, common user maybe won't find it.
https://github.com/utterance/utterances/tree/master/src/stylesheets/themes

File: layout/_third-party/comments/utteranc.swig (hunk 2 above)
Comment (1v9) [reply layout/_third-party/comments/utteranc.swig]: We use 2 spaces indentation 🚀

File: _config.yml [hunk 3]
```diff
@@ -679,6 +679,14 @@ gitalk:
...
+utteranc:
+  enable: false
+  repo:  #Github repo such as :TrumanDu/comments
+  pathname:  pathname
```
Comment (1v9) [inline _config.yml]: use `issue-term`, `pathname` is one value of this option.

Comment (1v9) [reply _config.yml]: Also you need to make a simple explanation for some option like this or leave an useful link.

File: layout/_third-party/comments/utteranc.swig (hunk 2 above)
Comment (ivan-nginx) [reply layout/_third-party/comments/utteranc.swig]: I think no need `https` here. This not google, right?

File: _config.yml (hunk 3 above)
Comment (TrumanDu) [reply _config.yml]: u have got it.add document in [here](https://github.com/theme-next/theme-next.org/pull/55/files)

**Comment by jiangtj** [comment]
wow, utterances  maybe better than gitalk

**Comment by jiangtj** [comment]
Emmm, there are some problems when the scheme is Gemini.
image

**Comment by 1v9** [comment]
> Emmm, there are some problems when the scheme is Gemini.
> image

脱缰野马😂

**Comment by jiangtj** [comment]
Besides, you can also try to make it use as a NexT plugin.

e.g, copy this [utterance.js](https://github.com/jiangtj/hexo-theme-next/blob/utteranc-try/scripts/filters/comment/utterance.js) to your hexo `scripts`, it will work as you expected.

If you want to make a NexT plugin, here are some examples to refer to.
https://github.com/theme-next/awesome-next#next-plugins   next-plugins sections.
And then, submit PR to awesome-next 😄

**Comment by jiangtj** [comment]
Done in here. Thank you for your work. https://github.com/theme-next/hexo-next-utteranc

**Comment by jiangtj** [comment]
@all-contributors please add @TrumanDu for code

**Pull request closed without merging**